File: AOC 2022/Python drafts/Day4Part1.py
#pre-processing
file = open("aoc22day4.txt",'r')
data = file.read()
file.close()

counter = 0
pairs = data.split("\n");
elves = []
for pair in pairs:
  elves += [pair.split(',')]

for elf in elves:
  elf1 = elf[0].split("-")
  elf2 = elf[1].split("-")

  #if elf1 fully encompasses elf2
  if (int(elf1[0])<=int(elf2[0]) and int(elf1[1])>=int(elf2[1])):
    counter += 1

  #if elf2 fully encompasses elf1
  elif (int(elf2[0]) <= int(elf1[0]) and int(elf2[1])>=int(elf1[1])):  
    counter += 1

print(counter)

File: AOC 2022/C++ solutions/Day4Part1.cpp
#include<iostream>
#include <fstream>
using namespace std;

int main(){
  string s;
  ifstream dataFile("aoc22day4.txt");
  int counter = 0;
  
  //also incomplete as I'm randomly having stoi errors in the middle of going through the data
  while (getline(dataFile, s)) {
    //every new pair per line
    int i = s.find(','); 
    int j = s.substr(0,i).find('-');
    int elf1[] = {stoi(s.substr(0,j)), stoi(s.substr(j+1,i))};
    
    j = s.substr(i+1,s.length()).find('-');
    int elf2[] = {stoi(s.substr(i+1,j)), stoi(s.substr(j+1,s.length()))};

    //cout <<elf1[0]<<"   "<<elf1[1]<<endl;
    //cout <<elf2[0]<<"   "<<elf2[1]<<endl;
    
    if (elf1[0]<=elf2[0]){ //elf1 potentially fully contains elf2
      if (elf1[1]>=elf2[1]){
        //fully contains 
        counter += 1;
        //cout << elf1<<elf2<<endl;
      }
    }
    else if (elf2[0]<=elf1[0]){ //elf2 might fully contain elf1
      if (elf2[1]>=elf1[1]){
        //fully contains
        counter += 1;
        //cout << elf2<<elf1 <<endl;
      }
    }
    
  }
  cout << counter << endl;

}